### ISSUE TITLE
Sharding / Shard Key

### ISSUE THREAD

**Issue by bradenwright** [carrierwaveuploader/carrierwave-mongoid#141]
Want//Need to enable sharding on fs.files and fs.chunks.  Basically its creating a bottleneck when I upload/process too many files.  Mongoid supports doing so but not sure how to make this work with carrierwave-mongoid which is what I use to upload my files into gridfs.  I'm open on how to do this but here is a little more detail.
 
fs.files:
{
  "_id" : <ObjectId>,
  "length" : <num>,
  "chunkSize" : <num>,
  "uploadDate" : <timestamp>,
  "md5" : <hash>,

  "filename" : <string>,
  "contentType" : <string>,
  "aliases" : <string array>,
  "metadata" : <dataObject>,
}

fs.chunks:
{
  "_id" : <ObjectId>,
  "files_id" : <ObjectId>,
  "n" : <num>,
  "data" : <binary>
}

So basically have 2 questions/issues related to implementing:
1) mongoid documentation mentions adding a "shard key" field to the model ( http://mongoid.org/en/mongoid/docs/installation.html#sharding ) so how would I do that for gridfs files since I don't have a model to do this on.
2) mongodb documentation mentions the _id fields increase sequentially so using _id or file_id are not good shard keys unless a custom id (non-sequential) is used ( http://docs.mongodb.org/manual/tutorial/shard-gridfs-data/ )

Thanks for any help, really just looking for random distribution so I'm open to using whatever custom id, shard keys, etc.  Doesn't have to be what I've mentioned

**Comment by bradenwright** [comment]
Looks like I was able to get it working at least....  I still didn't have a way to provide a shard key (as I do at the model level).  So not sure if its best practice or not but its working.

I decided to used hashed shard keys on fs.files._id and fs.chunks.files_id.  (So its evenly distributed)

If there are recommendation/comments I'm open

**Comment by rmm5t** [comment]
These questions should be directed towards https://github.com/ahoward/mongoid-grid_fs instead. 

Note, mongoid-grid_fs does create a modl for the files and chunks.
